### PR TITLE
Update Rust dependencies

### PR DESCRIPTION
**Pull request by tugunchinov**
For the #646

### CODE REVIEW

File: rust/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 /target/
+/lib/Cargo.lock
```
Comment (ia0) [inline rust/.gitignore:2]: Lock files are checked in for reproducibility purposes.

See https://github.com/rust-lang/cargo/blob/master/src/doc/src/faq.md#why-have-cargolock-in-version-control but this is indeed a contentious point in the Rust community and there are not yet clear guidelines. Note though that `cargo new` changed behavior to check the lock file, so this is the current default.

**Review by ia0 — CHANGES_REQUESTED** [review]
Thanks! I'm happy to approve the dependency requirements update. But I wish to keep the lock file for reproducibility.

File: rust/.gitignore (hunk 1 above)
Comment (tugunchinov) [reply rust/.gitignore]: Okay, I think I had outdated information about keeping `Cargo.lock`. Thank you, I'll refresh my brain's database. 

Anyway, I pushed the commit with dependencies updates only.

**Review by ia0 — COMMENTED** [review]
Thanks! You'll need to run `./test.sh` from the `rust` directory to update the CLI lock file too and make sure the CI passes.

**Comment by tugunchinov** [comment]
Done 🫡

**Review by ia0 — APPROVED** [review]
Thanks! Running the CI.

**Pull request merged**